openmdao/solvers/solver.py

```diff
@@ -365,12 +372,21 @@ def _iter_initialize(self):
             error at the first iteration.
         """
         if self.options['maxiter'] > 0:
+            self._run_apply()
             norm = self._iter_get_norm()
         else:
             norm = 1.0
         norm0 = norm if norm != 0.0 else 1.0
         return norm0, norm
 
+    def _run_apply(self):
+        """
+        Run the the apply_nonlinear method on the system.
+        """
+        recording_iteration_stack.append(('_run_apply', 0))
```

Comment:
Is it necessary to push '_run_apply' onto the iteration stack when there is no recording associated with _run_apply and we already do recording inside of _apply_nonlinear (and in _apply_linear)?